File: EscPosSharp/Image/CoffeeImageImpl.cs
using System.Drawing;

namespace EscPosSharp.Image;

/// <summary>
/// implements CoffeeImage using System.Drawing.Bitmap
/// </summary>
/// <remarks>
/// @seeCoffeeImage
/// @seeBufferedImage
/// </remarks>
public class CoffeeImageImpl : ICoffeeImage
{
    protected Bitmap image;

    public CoffeeImageImpl(Bitmap image)
    {
        this.image = image;
    }

    public virtual int GetWidth()
    {
        return image.Width;
    }

    public virtual int GetHeight()
    {
        return image.Height;
    }

    public virtual ICoffeeImage GetSubimage(int x, int y, int w, int h)
    {
        var newImage = new Bitmap(w, h);
        using var graphics = Graphics.FromImage(newImage);
        graphics.DrawImage(
            image,
            new Rectangle(0, 0, w, h),
            new Rectangle(x, y, w, h),
            GraphicsUnit.Pixel
        );
        return new CoffeeImageImpl(newImage);
    }

    public virtual int GetARGB(int x, int y)
    {
        return image.GetPixel(x, y).ToArgb();
    }
}

File: src/main/java/com/github/anastaciocintra/escpos/image/CoffeeImageImpl.java
using Java.Awt.Image;
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.Text;
using static Com.Github.Anastaciocintra.Escpos.Image.CharacterCodeTable;
using static Com.Github.Anastaciocintra.Escpos.Image.CutMode;
using static Com.Github.Anastaciocintra.Escpos.Image.PinConnector;
using static Com.Github.Anastaciocintra.Escpos.Image.Justification;
using static Com.Github.Anastaciocintra.Escpos.Image.FontName;
using static Com.Github.Anastaciocintra.Escpos.Image.FontSize;
using static Com.Github.Anastaciocintra.Escpos.Image.Underline;
using static Com.Github.Anastaciocintra.Escpos.Image.ColorMode;
using static Com.Github.Anastaciocintra.Escpos.Image.BarCodeSystem;
using static Com.Github.Anastaciocintra.Escpos.Image.BarCodeHRIPosition;
using static Com.Github.Anastaciocintra.Escpos.Image.BarCodeHRIFont;
using static Com.Github.Anastaciocintra.Escpos.Image.PDF417ErrorLevel;
using static Com.Github.Anastaciocintra.Escpos.Image.PDF417Option;
using static Com.Github.Anastaciocintra.Escpos.Image.QRModel;
using static Com.Github.Anastaciocintra.Escpos.Image.QRErrorCorrectionLevel;
using static Com.Github.Anastaciocintra.Escpos.Image.BitImageMode;

namespace Com.Github.Anastaciocintra.Escpos.Image
{
    /// <summary>
    /// implements CoffeeImage using Java BufferedImage
    /// </summary>
    /// <remarks>
    /// @seeCoffeeImage
    /// @seeBufferedImage
    /// </remarks>
    public class CoffeeImageImpl : CoffeeImage
    {
        protected BufferedImage image;
        public CoffeeImageImpl(BufferedImage image)
        {
            this.image = image;
        }

        public virtual int GetWidth()
        {
            return image.GetWidth();
        }

        public virtual int GetHeight()
        {
            return image.GetHeight();
        }

        public virtual CoffeeImage GetSubimage(int x, int y, int w, int h)
        {
            return new CoffeeImageImpl(image.GetSubimage(x, y, w, h));
        }

        public virtual int GetRGB(int x, int y)
        {
            return image.GetRGB(x, y);
        }
    }
}